src/main/java/com/cloudbees/jenkins/plugins/customtools/DecoratedLauncher.java

```diff
@@ -1,95 +0,0 @@
-/*
- * Copyright 2012, CloudBees Inc., Synopsys Inc. and contributors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.cloudbees.jenkins.plugins.customtools;
-
-import hudson.FilePath;
-import hudson.Launcher;
-import hudson.Proc;
-import hudson.model.Computer;
-import hudson.model.TaskListener;
-import hudson.remoting.Channel;
-
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStream;
-import java.util.Map;
-
-/**
- * A launcher which delegates to a provided inner launcher.
- * Allows subclasses to only implement methods they want to 
- * override.
- * 
- * @author rcampbell
- * @author Oleg Nenashev
- */
-//TODO: 1.567 provides a replacement
-public class DecoratedLauncher extends Launcher {
```

Comment:
It is a part of public plugin API, you cannot just remove it 🐛 

Instead of that, I recommend to....

* Make the class extend the core's implementation, remove overrides when they replicate the behavior from the parent class
* Mark the class as `@Deprecated @Restricted(NoUse.class) @RestrictedSince("TODO")`